lib/kramdown/converter/syntax_highlighter/rouge.rb

```diff
@@ -17,17 +17,23 @@ module Rouge
 
       # Highlighting via Rouge is available if this constant is +true+.
       AVAILABLE = true
+
+      # Rouge::Formatters::HTMLLegacy is available on Rouge 2.0 or later
+      FORMATTER_CLASS = ::Rouge::Formatters::HTMLLegacy
     rescue LoadError, SyntaxError
       AVAILABLE = false  # :nodoc:
+    rescue NameError
```

Comment:
What I meant was putting the whole `begin ... rescue ... end` block inside the `begin...rescue` block where availability is checked. Then it is clear which exception handling mechanism handles which situation.